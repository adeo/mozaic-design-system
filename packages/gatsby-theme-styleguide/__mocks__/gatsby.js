const React = require(`react`)
const gatsby = jest.requireActual(`gatsby`)

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn(),
}
