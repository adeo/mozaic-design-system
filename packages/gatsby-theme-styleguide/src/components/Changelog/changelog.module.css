/* Changelog */

.changelogContainer h2 {
  margin-top: 2em;
}

.changelogContainer h2,
.changelogContainer h3 {
  margin-bottom: 1em;
}

.changelogContainer h3 {
  margin-top: 1.5em;
  line-height: 1;
}

.changelogItem {
  padding-bottom: 2em;
  border-bottom: solid 1px #c3c3c3;
}
