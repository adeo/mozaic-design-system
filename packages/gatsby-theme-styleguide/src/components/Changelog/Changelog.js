import * as React from 'react'
import { useStaticQuery, graphql } from 'gatsby'
import * as styles from './changelog.module.css'

const Changelog = () => {
  const data = useStaticQuery(graphql`
    query {
      allGithubRelease {
        edges {
          node {
            descriptionHTML
          }
        }
      }
    }
  `)

  return (
    <div className={styles.changelogContainer}>
      {data.allGithubRelease.edges
        .filter((release) => release.node.descriptionHTML !== '')
        .map((release, i) => (
          <section
            className={styles.changelogItem}
            key={`release-${i}`}
            first={i === 0}
            dangerouslySetInnerHTML={{ __html: release.node.descriptionHTML }}
          />
        ))}
    </div>
  )
}

export default Changelog
