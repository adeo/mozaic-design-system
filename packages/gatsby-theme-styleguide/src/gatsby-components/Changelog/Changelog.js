import React from 'react'
import styled, { css } from 'styled-components'
import { StaticQuery, graphql } from 'gatsby'

const ChangelogContainer = styled.div`
  h2 {
    margin-top: 2em;
  }

  h2,
  h3 {
    margin-bottom: 1em !important;
  }

  h3 {
    margin-top: 1.5em !important;
    line-height: 1;
  }
`

export const ChangelogItem = styled.div`
  ${({ first }) =>
    !first &&
    css`
      font-size: 0.8rem;
    `}
  padding-bottom: 2em;
  border-bottom: solid 1px #c3c3c3;
`

const ChangelogComponent = ({ data }) => (
  <ChangelogContainer>
    {data.allGithubRelease.edges
      .filter(release => release.node.descriptionHTML !== '')
      .map((release, i) => (
        <ChangelogItem
          key={`release-${i}`}
          first={i === 0}
          dangerouslySetInnerHTML={{ __html: release.node.descriptionHTML }}
        />
      ))}
  </ChangelogContainer>
)

const query = graphql`
  query AllGitReleasesQuery2 {
    allGithubRelease {
      edges {
        node {
          descriptionHTML
        }
      }
    }
  }
`

const Changelog = () => (
  <StaticQuery
    query={query}
    render={data => <ChangelogComponent data={data} />}
  />
)

export default Changelog
