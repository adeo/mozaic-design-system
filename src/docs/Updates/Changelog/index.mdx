---
title: 'Changelog'
order: 1
description: 'The changelog that contain all changes for every Mozaic releases'
---

import Changelog from '@mozaic-ds/gatsby-theme-styleguide/src/gatsby-components/Changelog'

<Changelog />
