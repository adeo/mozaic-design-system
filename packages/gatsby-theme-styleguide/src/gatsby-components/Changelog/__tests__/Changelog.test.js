import React from 'react'
import renderer from 'react-test-renderer'
import { StaticQuery } from 'gatsby'

import Changelog, { ChangelogItem } from '../Changelog'

const data = {
  allGithubRelease: {
    edges: [
      {
        node: {
          descriptionHTML: '<div>v1.0.1-alpha.31</div>',
        },
      },
      {
        node: {
          descriptionHTML: '<div>v1.0.1-alpha.30</div>',
        },
      },
      {
        node: {
          descriptionHTML: '',
        },
      },
      {
        node: {
          descriptionHTML: '<div>v1.0.1-alpha.28</div>',
        },
      },
      {
        node: {
          descriptionHTML: '<div>v1.0.1-alpha.26</div>',
        },
      },
    ],
  },
}

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) => render(data))
})

describe('<Changelog />', () => {
  it('render corectly', () => {
    const tree = renderer.create(<Changelog />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('do not render releases with empty descriptions', () => {
    const tree = renderer.create(<Changelog />)
    expect(tree.root.findAllByType(ChangelogItem).length).toEqual(4)
  })
})
