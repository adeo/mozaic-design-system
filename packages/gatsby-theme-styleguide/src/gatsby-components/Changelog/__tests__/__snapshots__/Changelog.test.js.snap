// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Changelog /> render corectly 1`] = `
.c0 h2 {
  margin-top: 2em;
}

.c0 h2,
.c0 h3 {
  margin-bottom: 1em !important;
}

.c0 h3 {
  margin-top: 1.5em !important;
  line-height: 1;
}

.c1 {
  padding-bottom: 2em;
  border-bottom: solid 1px #c3c3c3;
}

.c2 {
  font-size: 0.8rem;
  padding-bottom: 2em;
  border-bottom: solid 1px #c3c3c3;
}

<div
  className="c0"
>
  <div
    className="c1"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<div>v1.0.1-alpha.31</div>",
      }
    }
  />
  <div
    className="c2"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<div>v1.0.1-alpha.30</div>",
      }
    }
  />
  <div
    className="c2"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<div>v1.0.1-alpha.28</div>",
      }
    }
  />
  <div
    className="c2"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<div>v1.0.1-alpha.26</div>",
      }
    }
  />
</div>
`;
